src/reduxForm.js

```diff
@@ -119,7 +119,7 @@ const createReduxForm =
             if (nextProps) {
               const { enableReinitialize } = this.props
               if ((enableReinitialize || !nextProps.initialized) && !deepEqual(this.props.initialValues, nextProps.initialValues)) {
-                this.props.initialize(nextProps.initialValues)
+                this.props.initialize(nextProps.initialValues, true)
```

Comment:
I'm pretty sure this should default to `false`. Perhaps there could be a `keepDirtyOnInitialize` config parameter or something?